src/main_loop.rs

```diff
@@ -158,13 +159,14 @@ impl Future for MainLoopState {
                 self.librespot_connection.connection = Box::pin(futures::future::pending());
                 let backend = self.audio_setup.backend;
                 let audio_device = self.audio_setup.audio_device.clone();
+                let audio_format = self.audio_setup.audio_format;
                 let (player, event_channel) = Player::new(
                     self.player_config.clone(),
                     session.clone(),
                     audio_filter,
                     // TODO: dunno how to work with AudioFormat yet, maybe dig further if this
                     // doesn't work for all configurations
-                    move || (backend)(audio_device, AudioFormat::default()),
+                    move || (backend)(audio_device, audio_format),
```

Comment:
I guess, we can remove the comment now that you've hit such a case where it did not work and changed it accordingly :-)